masonry/src/tree_arena.rs

```diff
@@ -62,12 +33,28 @@ pub struct TreeArena<Item> {
     parents_map: HashMap<u64, Option<u64>>,
 }
 
+/// A reference type giving shared access to an item and its children.
+///
+/// When you borrow an item from a [`TreeArena`], it returns an `ArenaRef`.
```

Comment:
Would it be worth pointing to a concrete method here (`TreeArena::find`?)

Reply:
I'm not sure about this, because `TreeArena::find` is more of a syntactic sugar than something your architecture is supposed to rest on.